src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -273,4 +277,12 @@ mkAgentMessage encKey senderTs agentMessage = do
               agentMessage
             }
   paddedSize <- asks paddedMsgSize
-  liftError CRYPTO $ C.encrypt encKey paddedSize msg
+  liftError cryptoError $ C.encrypt encKey paddedSize msg
+
+cryptoError :: C.CryptoError -> AgentErrorType
+cryptoError = \case
+  C.CryptoLargeMsgError -> CMD LARGE
```

Comment:
Is it secure to differentiate this error? Maybe use the same response for all crypto related errors? Not sure again, just a thought..

Reply:
nvm I guess, I was thinking of something like padding oracle attack, but this is not similar and doesn't really leak anything not already known